pkg/handler/handler.go

```diff
@@ -560,7 +560,7 @@ func (h handler) bind(w http.ResponseWriter, r *http.Request, params map[string]
 	}
 
 	// process binding request
-	resp, err := h.broker.Bind(serviceInstance, bindingUUID, req, async)
+	resp, ranAsync, err := h.broker.Bind(serviceInstance, bindingUUID, req, async)
```

Comment:
I don't think we need to return this boolean. We know if we are async based on the async. `Provision` returns similar values without returning a `ranAsync`. I'd prefer to model it similar to `Provision`.

```golang
    if err != nil {
        switch err {
        case broker.ErrorDuplicate:
            writeResponse(w, http.StatusConflict, broker.ProvisionResponse{})
        case broker.ErrorProvisionInProgress:
            writeResponse(w, http.StatusAccepted, resp)
        case broker.ErrorAlreadyProvisioned:
            writeResponse(w, http.StatusOK, resp)
        case broker.ErrorNotFound:
            writeResponse(w, http.StatusBadRequest, broker.ErrorResponse{Description: err.Error()})
        default:
            writeResponse(w, http.StatusBadRequest, broker.ErrorResponse{Description: err.Error()})
        }
    } else if async {
        writeDefaultResponse(w, http.StatusAccepted, resp, err)
    } else {
        writeDefaultResponse(w, http.StatusCreated, resp, err)
    }
```

Reply:
When ``async == true`` for the argument being passed into the broker's Bind function, there are at least two cases where there will not be asynchronous execution.

1. if the binding already exists
2. if the broker is configured for ``launch_apb_on_bind = false``

In theory there could be other reasons for doing it synchronously even if ``async == true``.

Put another way, when ``async == true``, all that really means is that the client specified ``accepts_incomplete=true``. That is not a guarantee that asynchronous execution will happen. The handler must allow the broker to decide whether async execution is appropriate, and must then compose an http response based on what the broker decided. And the only way for the handler to know what the broker decided is if the broker tells it, hence the ``bool``.

A different but perfectly valid (I think?) way to handle this would be to add an attribute to the response object and communicate through that. But since the response object gets directly serialized into an HTTP response body, that might be overloading its definition.

We could add a private field "async" to the response, and then a method that returns whether the attribute was async or not, and I assume that would allow the field to be omitted from serialization to JSON? Something like:

```
type BindResponse struct {
	Credentials     map[string]interface{} `json:"credentials,omitempty"`
	SyslogDrainURL  string                 `json:"syslog_drain_url,omitempty"`
	RouteServiceURL string                 `json:"route_service_url,omitempty"`
	VolumeMounts    []interface{}          `json:"volume_mounts,omitempty"`
	Operation       string                 `json:"operation,omitempty"`
        ranAsync        bool
}

// returns true if the operation executed asynchronously, otherwise false
func (b BindResponse) RanAsync() (bool)
        return b.ranAsync
```

Even if that's doable, it's probably simpler, and definitely just as effective, to return a separate value.

Reply:
@mhrivnak very good explanation to why you did it differently than provision. Okay that makes sense now. Sorry I missed that.